src/domain/opportunity/opportunity.service.ts

```diff
@@ -224,6 +228,32 @@ export class OpportunityService {
     return newActorGroup;
   }
 
+  async createProject(
+    opportunityId: number,
+    projectData: ProjectInput
+  ): Promise<IProject> {
+    const opportunity = await this.getOpportunityByID(opportunityId);
+    if (!opportunity)
+      throw new Error(`Unalbe to locate opportunity with id: ${opportunityId}`);
+
+    // Check that do not already have an Project with the same name
+    const title = projectData.name;
+    const existingProject = opportunity.projects?.find(
```

Comment:
As far as the mockups go projects have Short Name (2 words max)? Should there be validation for its uniqueness as well?

Reply:
Yes, will raise a separate issue